src/components/znkExercise/utility/service/znkExerciseUtilitySrv.js

```diff
@@ -37,6 +37,23 @@
 
                 return !!isCorrect;
             };
+
+            ZnkExerciseUtilitySrv.setQuestionsGroupData = function (questions, groupData) {
+                var groupDataMap = {};
+
+                angular.forEach(groupData, function (group) {
+                    groupDataMap[group.id] = group;
+                });
+
+                angular.forEach(questions, function (question) {
+                    if (!groupDataMap[question.groupDataId]) {
+                        $log.debug('Group data is missing for the following question id ' + question.id);
```

Comment:
mmm, i think it's like in toefl when there's no groupData it's just log that group data is missing